GetJobber/atlantis
# feat(components): add modal size and child render support to confirmation modal

## Motivations
We had to use the <ConfirmationModal> component to display a warning to the user.
The message that had to be displayed could be fit in a `string` or `Markdown` format. We had to use JSX element. However, <ConfirmationModal> does not support displaying of the `children` in the component.

The confirmed design of the data that we need to display had to use a bigger size of the modal to fit all text nicely. But <ConfirmationModal> didn't have the ability to dynamically change the size of the displayed modal. This PR adds this functionality.

## Changes
- With the current proposal, <ConfirmationModal> can display either provided `message` or `children` component.
- Support for the modal size has been added.



### Added
- Added 2 new TS interfaces that must enforce displaying of information on the component


### Changed
- Updated `ConfirmationModalProps` structure to include support of the children component


### Deprecated
- n/a


### Removed
- n/a


### Fixed
- n/a


### Security
- n/a


## Testing
Added additional test that makes sure that either the `message` or `children` component is displayed.

1. Pass different modal sizes and make sure modal changes in size.
2. Try to add `message` and `children` props to the <ConfirmationModal> component. You should see TS error that won't let you do it.
3. Remove `message` prop from and make sure `children` are displayed.
3. Add back `message` prop and remove `children`. Make sure that `message` is displayed.


---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
